File: src/main/java/org/bk/graphql/config/DataLoadersConfig.java
package org.bk.graphql.config;

import org.bk.graphql.application.BookAuthorManagementService;
import org.bk.graphql.web.BookAuthorsDataLoader;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class DataLoadersConfig {

    @Bean
    public BookAuthorsDataLoader bookAuthorsDataLoader(BookAuthorManagementService bookAuthorManagementService) {
        return new BookAuthorsDataLoader(bookAuthorManagementService);
    }
}

File: src/main/kotlin/org/bk/graphql/config/DataLoadersConfig.kt
package org.bk.graphql.config

import org.bk.graphql.service.BookService
import org.bk.graphql.web.BookAuthorsDataLoader
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class DataLoadersConfig {

    @Bean
    fun bookAuthorsDataLoader(bookService: BookService): BookAuthorsDataLoader {
        return BookAuthorsDataLoader(bookService)
    }
}